metis/clt_rand.go

```diff
@@ -0,0 +1,68 @@
+package metis
+
+import (
+	"errors"
+	"math/rand"
+)
+
+var (
+	ErrOverflow     = errors.New("value is too large")
+	ErrInvalidRange = errors.New("invalid range")
+)
+
+const (
+	maxInt63 = int64(1<<62 - 1)
+)
+
+// CLTGenerator generates random integers that follow an approximate normal distribution
+type CLTGenerator struct{}
+
+// NewCLTGenerator initializes a CLT generator without state dependency
+func NewCLTGenerator() *CLTGenerator {
+	return &CLTGenerator{}
+}
+
+// GenerateRandomInt generates a random integer from min to max (inclusive), based on a provided seed.
+// This simulates a normal distribution by summing multiple uniform random variables.
+func (g *CLTGenerator) GenerateRandomInt(seed int64, min, max uint64) (int64, error) {
```

Comment:
Is there any difference from randomRangeInclusive?

Reply:
@aiden069 told me that the numbers generated by randomRangeInclusive is not in a normal distribution manner, so I rewrote the random generation part with CLT (since this is the only stateless random generation method I can find that will most likely provide a normal distribution result). 

Here is the test result:

![telegram-cloud-photo-size-5-6073483992117657275-y](https://github.com/user-attachments/assets/4e0d352c-b815-4439-957f-c66abdc44022)

Reply:
Sure, shall we setup a quick meeting for this? And @aiden069 what's your opinion here?